cli/src/main/java/org/crsh/cli/completers/EnumCompleter.java

```diff
@@ -41,7 +41,7 @@ public static EnumCompleter getInstance() {
   }
 
   public Completion complete(ParameterDescriptor parameter, String prefix) throws Exception {
-    if (parameter.getType() == ValueType.ENUM) {
+    if (ValueType.ENUM.equals(parameter.getType())) {
```

Comment:
Changing this equality evaluation (and others like it) seems to be arbitrary and it does not make the operation any safer.  And it loses some aesthetic appeal with using the equals method.  ValueType instances are static and final, so there should be no problem with equality performed this way.  Maybe reversing the sides of the equation would be better, so that "if (ValueType.ENUM) == parameter.getType())" won't throw a NPE if parameter, or its "type" field is null.  The only issue that could come from this equality comparison would be from classloader issues, if that would even be possible in this situation, but how could an equals() method even account for this and avoid problems?  Checkstyle and PMD warnings are not always valuable, and they should not unquestionably dictate your coding.  As with any automated tool, it is better to use the advice and also determine if it is sensible or preferable to implement it the suggested way.

Also, Object's equals method performs the same operation, as noted in the documentation: "The equals method for class Object implements the most discriminating possible equivalence relation on objects; that is, for any non-null reference values x and y, this method returns true if and only if x and y refer to the same object (x == y has the value true)."